packages/ui-react/src/components/Modal/Modal.tsx

```diff
@@ -16,74 +16,16 @@ type Props = {
   animationContainerClassName?: string;
 } & React.AriaAttributes;
 
-const Modal: React.FC<Props> = ({ open, onClose, children, AnimationComponent = Grow, animationContainerClassName, ...ariaAtributes }: Props) => {
-  const [visible, setVisible] = useState(open);
-  const lastFocus = useRef<HTMLElement>() as React.MutableRefObject<HTMLElement>;
+const Modal: React.FC<Props> = ({ open, onClose, children, AnimationComponent = Grow, animationContainerClassName, ...ariaAttributes }: Props) => {
   const modalRef = useRef<HTMLDivElement>() as React.MutableRefObject<HTMLDivElement>;
 
-  const keyDownEventHandler = (event: React.KeyboardEvent) => {
-    if (event.key === 'Escape' && onClose) {
-      onClose();
-    }
-  };
-
-  // delay the transition state so the CSS transition kicks in after toggling the `open` prop
-  useEffect(() => {
-    const activeElement = document.activeElement as HTMLElement;
-    const appView = document.querySelector('#root') as HTMLDivElement;
-
-    if (open) {
-      // store last focussed element
-      if (activeElement) {
-        lastFocus.current = activeElement;
-      }
-
-      // reset the visible state
-      setVisible(true);
-
-      // make sure main content is hidden for screen readers and inert
-      if (appView) {
-        appView.inert = true;
-      }
-
-      // prevent scrolling under the modal
-      document.body.style.marginRight = `${scrollbarSize()}px`;
-      document.body.style.overflowY = 'hidden';
-    } else {
-      if (appView) {
-        appView.inert = false;
-      }
-
-      document.body.style.removeProperty('margin-right');
-      document.body.style.removeProperty('overflow-y');
-    }
-  }, [open]);
-
-  useEffect(() => {
-    if (visible) {
```

Comment:
@langemike @MelissaDTH the `visible` state is not needed in the browser, do you think it is still needed for screen readers? 🤔

Reply:
As we discussed during this week, I had tested this on a screen reader (iOS) and saw that the focus was not correctly set on the first element in the modal after switching between modals. 

But, when a modal is initially opened, the focus **is** correctly set on the first element (the email input field e.g.). So I think that the focus that is lost between modal switching is caused by something different. So that makes me think that this can be safely removed?